hopp/simulation/technologies/resource/nsrdb_data.py

```diff
@@ -0,0 +1,226 @@
+        super().__init__(lat, lon, year)
+
+        if filepath == "" and nsrdb_source_path=="":
+            # use default filepath
+            self.nsrdb_file = NSRDB_NEW + "{}.h5".format(self.year)
+        elif filepath != "" and nsrdb_source_path == "":
+            # filepath (full h5 filepath) is provided by user
+            if ".h5" not in filepath:
+                filepath = filepath + ".h5"
+            self.nsrdb_file = filepath
+        elif filepath == "" and nsrdb_source_path != "":
+            # directory of h5 files (nsrdb_source_path) is provided by user
+            self.nsrdb_file = os.path.join(str(nsrdb_source_path),"nsrdb_{}.h5".format(self.year))
+        else:
+            # use default filepaths
+            self.nsrdb_file = NSRDB_NEW + "{}.h5".format(self.year)
+        
+        # Check for valid year
+        if self.year < 1998 or self.year > 2022:
+                raise ValueError(f"Resource year for NSRDB Data must be between 1998 and 2022 but {self.year} was provided")
+        
+        # Check for valid filepath for NSRDB file
+        if not os.path.isfile(self.nsrdb_file):
+            raise FileNotFoundError(f"Cannot find NSRDB .h5 file, filepath {self.nsrdb_file} does not exist")
+
+        # Pull data from HPC NSRDB dataset
+        self.download_resource()
+
+        # Set solar resource data into SAM/PySAM digestible format
+        self.format_data()
+        
+
+    def download_resource(self):
+        """load NSRDB h5 file using rex and get solar resource data for location
+        specified by (self.lat, self.lon)
+        """
+        
+        # Open file with rex NSRDBX object
+        with NSRDBX(self.nsrdb_file, hsds=False) as f:
+            # get gid of location closest to given lat/lon coordinates
+            site_gid = f.lat_lon_gid((self.latitude,self.longitude))
+
+            # extract timezone, elevation, latitude and longitude from meta dataset with gid
+            self.time_zone = f.meta['timezone'].iloc[site_gid]
+            self.elevation = f.meta['elevation'].iloc[site_gid]
+            self.nsrdb_latitude = f.meta['latitude'].iloc[site_gid]
+            self.nsrdb_longitude = f.meta['longitude'].iloc[site_gid]
+            
+            # extract remaining datapoints: year, month, day, hour, minute, dn, df, gh, wspd,tdry, pres, tdew
+            # NOTE: datasets have readings at 0 and 30 minutes each hour, HOPP/SAM workflow requires only 30 minute reading values -> filter 0 minute readings with [1::2]
+            # NOTE: datasets are not auto shifted by timezone offset -> wrap extraction in SAMResource.roll_timeseries(input_array, timezone, #steps in an hour=1) to roll timezones
+            # NOTE: solar_resource.py code references solar_zenith_angle and RH = relative_humidity but I couldn't find them actually being utilized. Captured them below just in case.
```

Comment:
Could you move these extra long comments to multiple lines? I know we don't have any enforcement on this, but it's nice to be able to read comments without sideways scrolling.